apps/heft/src/plugins/CopyFilesPlugin.ts

```diff
@@ -0,0 +1,336 @@
+          onlyFiles: true
+        })
+      );
+
+      // Dedupe and throw if a double-write is detected
+      for (const destinationFolderRelativePath of copyConfiguration.destinationFolders) {
+        for (const sourceFileRelativePath of sourceFileRelativePaths) {
+          // Only include the relative path from the sourceFolder if flatten is false
+          const resolvedSourceFilePath: string = path.join(resolvedSourceFolderPath, sourceFileRelativePath);
+          const resolvedDestinationFilePath: string = path.resolve(
+            buildFolder,
+            destinationFolderRelativePath,
+            copyConfiguration.flatten ? '.' : path.dirname(sourceFileRelativePath),
+            path.basename(sourceFileRelativePath)
+          );
+
+          // Throw if a duplicate copy target with a different source or options is specified
+          const existingDestinationCopyDescriptor:
+            | ICopyFileDescriptor
+            | undefined = destinationCopyDescriptors.get(resolvedDestinationFilePath);
+          if (existingDestinationCopyDescriptor) {
+            if (
+              existingDestinationCopyDescriptor.sourceFilePath === resolvedSourceFilePath &&
+              existingDestinationCopyDescriptor.hardlink === !!copyConfiguration.hardlink
+            ) {
+              // Found a duplicate, avoid adding again
+              continue;
+            }
+            throw new Error(
+              `Cannot copy different files to the same destination "${resolvedDestinationFilePath}"`
+            );
+          }
+
+          // Finally, add to the map and default hardlink to false
+          let sourceCopyDescriptor: ICopyFileDescriptor | undefined = sourceCopyDescriptors.get(
+            resolvedSourceFilePath
+          );
+          if (!sourceCopyDescriptor) {
+            sourceCopyDescriptor = {
+              sourceFilePath: resolvedSourceFilePath,
+              destinationFilePaths: [resolvedDestinationFilePath],
+              hardlink: !!copyConfiguration.hardlink
+            };
+            sourceCopyDescriptors.set(resolvedSourceFilePath, sourceCopyDescriptor);
+          } else {
+            sourceCopyDescriptor.destinationFilePaths.push(resolvedDestinationFilePath);
+          }
+
+          // Add to other map to allow deduping
+          destinationCopyDescriptors.set(resolvedDestinationFilePath, sourceCopyDescriptor);
+        }
+      }
+    }
+
+    // We're done with the map, grab the values and return
+    return Array.from(sourceCopyDescriptors.values());
+  }
+
+  private _getIncludedGlobPatterns(copyConfiguration: IExtendedSharedCopyConfiguration): string[] {
+    // Glob extensions with a specific glob to increase perf
+    const patternsToGlob: Set<string> = new Set<string>();
+    for (const fileExtension of copyConfiguration.fileExtensions || []) {
+      const escapedExtension: string = glob.escapePath(fileExtension);
+      patternsToGlob.add(`**/*${escapedExtension}`);
+    }
```

Comment:
Would this:
```
**/*.{png,gif,jpg,jpeg,svg}
```
...be significantly more efficient than this?
```
**/*.png
**/*.gif
**/*.jpg
**/*.jpeg
**/*.svg
```
If so it might be worth optimizing them, since this should be the 99% most common case.

(Ideally we could later optimize it even further and skip invoking the glob engine entirely, if the pattern only specifies file extensions.)